### PR TITLE
Submissions

### PR DESCRIPTION
**Pull request by tsbolty**
### What this PR does (required):
- Can display all contest submissions to contest owner as well as display contest description to all. Also added a modal to display full sized image.

### Screenshots / Videos (required):
image

image

image

### Any information needed to test this feature (required):
- To see submissions, you need to log in as the owner of the contest

### Any issues with the current functionality (optional):
- Completed all styling and integration. Since I changed the click event of the image to show the modal, we will have to discuss how we want the UI to look to select a winner.

### CODE REVIEW

File: client/src/components/ImageModal/ImageModal.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  const handleOpen = () => {
+    setIsOpen(true);
+  };
+
+  return (
+    <div>
+      <div onClick={handleOpen}>{children}</div>
+      <Modal open={isOpen} onClose={handleClose} className={classes.modal}>
+        <Paper className={classes.body}>
+          <Box className={classes.box}>
```
Comment (rajivtitus) [inline client/src/components/ImageModal/ImageModal.tsx]: Material UI has flex-box built into it, you add props straight to the component here

File: client/src/pages/Contest/Contest.tsx [hunk 2]
```diff
@@ -134,27 +153,28 @@ export default function Contest(): JSX.Element {
           </Grid>
         </Grid>
         <Grid className={classes.grid} container>
-          <Grid item xs={10}>
+          <Grid item xs={12} sm={10}>
             <Typography className={classes.contestTitle}>
-              {contestSubmissions.title ? contestSubmissions.title : 'Lion tattoo concept in minimal style'}{' '}
+              {contest ? contest.title : 'Lion tattoo concept in minimal style'}{' '}
```
Comment (rajivtitus) [inline client/src/pages/Contest/Contest.tsx]: Can we please remove the hardcoded data in this component?
Ditto throughout

File: client/src/pages/Contest/Contest.tsx [hunk 3]
```diff
@@ -177,21 +197,54 @@ export default function Contest(): JSX.Element {
...
+                  </ImageList>
+                </div>
+              </Panel>
+            ) : null}
+            <Panel value={value} index={contestSubmissions.length ? 1 : 0}>
+              <Typography align="center" variant="h3" className={classes.descriptionHeader}>
+                Tattoo Description:
+              </Typography>
+              <Typography align="center" variant="h5">
+                {contest ? contest.description : 'Description of the contest'}
```
Comment (rajivtitus) [inline client/src/pages/Contest/Contest.tsx]: Same as above

File: client/src/pages/Contest/Contest.tsx [hunk 4]
```diff
@@ -177,21 +197,54 @@ export default function Contest(): JSX.Element {
...
               </Typography>
+              <ImageList cols={4} gap={10} className={classes.imageList}>
+                {contest &&
```
Comment (rajivtitus) [inline client/src/pages/Contest/Contest.tsx]: You can write this in a single line using a ternary in Javascript object

File: server/controllers/submission.js [hunk 5]
```diff
@@ -59,15 +59,28 @@ exports.getSubmissionByContest = asyncHandler(async (req, res) => {
...
+          images: item.images,
+          contest: item.contest,
+          active: item.active,
+          artistName: item.artistId.username,
+          artistPic: item.artistId.profilePic,
+          artistId: item.artistId._id,
+        }))
+        res.status(200).send(structured)
+      } else {
+        res.status(304).send([])
```
Comment (rajivtitus) [inline server/controllers/submission.js]: What is the reason for using a 304 status code here?

File: client/src/components/ImageModal/ImageModal.tsx (hunk 1 above)
Comment (tsbolty) [reply client/src/components/ImageModal/ImageModal.tsx]: Good call. This is fixed.

File: client/src/pages/Contest/Contest.tsx (hunk 2 above)
Comment (tsbolty) [reply client/src/pages/Contest/Contest.tsx]: Can do.

File: client/src/pages/Contest/Contest.tsx (hunk 3 above)
Comment (tsbolty) [reply client/src/pages/Contest/Contest.tsx]: Also done.

File: server/controllers/submission.js (hunk 5 above)
Comment (tsbolty) [reply server/controllers/submission.js]: The 304 was meant to communicate that the request went through successfully, but there were no results found and nothing was modified. After reading more into the 304 code, I believe a 204 would be a better choice here. What do you think?

Comment (rajivtitus) [reply server/controllers/submission.js]: Yes, the 204 makes a lot more sense here

**Review by rajivtitus — APPROVED** [review]
LGTM! Thanks for making requested changes

File: server/controllers/submission.js (hunk 5 above)
Comment (tsbolty) [reply server/controllers/submission.js]: Gotcha this is done.

File: client/src/pages/Contest/Contest.tsx (hunk 4 above)
Comment (tsbolty) [reply client/src/pages/Contest/Contest.tsx]: Done deal

**Pull request merged**